modules/references/server/controllers/reference.server.controller.js

```diff
@@ -186,30 +235,94 @@ exports.create = function (req, res, next) {
     function respond(savedReference, cb) {
       return cb({
         status: 201,
-        body: {
-          userFrom: savedReference.userFrom,
-          userTo: savedReference.userTo,
-          recommend: savedReference.recommend,
-          created: savedReference.created.getTime(),
-          met: savedReference.met,
-          hostedMe: savedReference.hostedMe,
-          hostedThem: savedReference.hostedThem,
-          id: savedReference._id,
-          public: savedReference.public
-        }
+        body: formatReference(savedReference)
       });
     }
-  ], function (err) {
-    // send error responses
-    if (err && err.status && err.body) {
-      if (err.body.errType) {
-        err.body.message = errorService.getErrorMessageByKey(err.body.errType);
-        delete err.body.errType;
+  ], processResponses.bind(this, res, next));
+};
+
+/**
+ * Validator for readMany controller
+ */
+function validateReadMany(req) {
```

Comment:
This request validation could be replaced with `json-schema` and `ajv` library. Or not. Or later.